### PR TITLE
Handle removed search indexers

### LINKED ISSUE #1252: Non-existent service error if search is disabled
Setting `contao.search.default_indexer.enable` to `false` result in an `You have requested a non-existent service "contao.search.indexer.delegating"` error when building the container.

### PR DESCRIPTION
**Pull request by Toflar**
Fixes #1252

### CODE REVIEW

File: core-bundle/src/DependencyInjection/Compiler/CrawlerPass.php [hunk 1]
```diff
@@ -16,20 +16,28 @@
...
 
     public function process(ContainerBuilder $container): void
     {
-        if (!$container->has('contao.crawl.escargot_factory')) {
+        $subscribers = $this->findAndSortTaggedServices('contao.escargot_subscriber', $container);
+
+        if (!$container->has('contao.crawl.escargot_factory') || 0 === \count($subscribers)) {
+
+            // Remove factory
+            $container->removeDefinition('contao.crawl.escargot_factory');
```
Comment (Toflar) [inline core-bundle/src/DependencyInjection/Compiler/CrawlerPass.php:29]: Not sure if this works. Maybe the has check needs to be extra but tests will show.

**Comment by Toflar** [comment]
Added unit tests. This is RTM. If you also remove the broken link checker, the crawler is now gone in the back end as is the `contao:crawl` command.

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/DependencyInjection/Compiler/CrawlerPass.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Compiler/CrawlerPass.php:29]: I am a bit confused. If the container does **not** have the definition, remove the definition?

File: core-bundle/src/DependencyInjection/Compiler/SearchIndexerPass.php [hunk 2]
```diff
@@ -28,31 +29,36 @@ class SearchIndexerPass implements CompilerPassInterface
...
             static function (Reference $reference): bool {
                 return self::DELEGATING_SERVICE_ID !== (string) $reference;
             }
         );
 
-        // Remove the service and the search index listener if there are no indexers
-        if (0 === \count($references)) {
+        if (!$container->hasDefinition(self::DELEGATING_SERVICE_ID) || 0 === \count($indexers)) {
+            // Remove delegating indexer
             $container->removeDefinition(self::DELEGATING_SERVICE_ID);
```
Comment (leofeyer) [inline core-bundle/src/DependencyInjection/Compiler/SearchIndexerPass.php:44]: Same here. Is this intentional?

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/DependencyInjection/Compiler/SearchIndexerPass.php (hunk 2 above)
Comment (Toflar) [reply core-bundle/src/DependencyInjection/Compiler/SearchIndexerPass.php:44]: Yes it is. It's either not there completely or there are no indexers/crawl subscribers. It's just abbreviated :)

File: core-bundle/tests/DependencyInjection/Compiler/CrawlerPassTest.php [hunk 3]
```diff
@@ -12,30 +12,47 @@
...
+        $this->assertTrue($container->hasDefinition('contao.command.crawl'));
+
         $methodCalls = $container->findDefinition('contao.crawl.escargot_factory')->getMethodCalls();
 
         $this->assertCount(1, $methodCalls);
         $this->assertSame('addSubscriber', $methodCalls[0][0]);
         $this->assertInstanceOf(Reference::class, $methodCalls[0][1][0]);
     }
+
+    public function testIfTheFactoryAndCrawlCommandAreRemovedWhenThereAreNoSubscribers(): void
```
Comment (leofeyer) [inline core-bundle/tests/DependencyInjection/Compiler/CrawlerPassTest.php]:
```suggestion
    public function testRemovesTheFactoryAndTheCrawlCommandIfThereAreNoSubscribers(): void
```

**Review by leofeyer — CHANGES_REQUESTED** [review]
Besides the one test method name, this PR is RTM IMHO.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**